src/_gettsim/transfers/erwerbsm_rente.py

```diff
@@ -0,0 +1,315 @@
+from _gettsim.piecewise_functions import piecewise_polynomial
+from _gettsim.shared import dates_active
+
+
+def erwerbsm_rente_m(
+    erwerbsm_rente_zugangsfaktor: float,
+    entgeltp_erwerbsm_rente: float,
+    rentenwert: float,
+    rentenartfaktor_erwerbsm_rente: float,
+    ges_rente_vorauss_erwerbsm: bool,
+) -> float:
+    """Calculating the Erwerbsminderungsrente pension claim
+
+    Legal reference: SGB VI § 64: Rentenformel für Monatsbetrag der Rente
+
+
+    Parameters
+    ----------
+    erwerbsm_rente_zugangsfaktor
+        See :func:`erwerbsm_rente_zugangsfaktor`.
+    entgeltp_erwerbsm_rente
+        See :func:`entgeltp_erwerbsm_rente`.
+    rentenwert
+        See :func:`rentenwert`.
+    rentenartfaktor_erwerbsm_rente
+        See :func:`rentenartfaktor_erwerbsm_rente`.
+    ges_rente_vorauss_erwerbsm
+        See :func:`ges_rente_vorauss_erwerbsm`.
+    Returns
+    -------
+    Erwerbsminderungsrente pension claim
+
+    """
+
+    if ges_rente_vorauss_erwerbsm:
+        out = (
+            entgeltp_erwerbsm_rente
+            * erwerbsm_rente_zugangsfaktor
+            * rentenwert
+            * rentenartfaktor_erwerbsm_rente
+        )
+    else:
+        out = 0.0
+    return out
+
+
+def ges_rente_vorauss_erwerbsm(
+    erwerbsgemindert: bool,
+    m_pflichtbeitrag: float,
+    ges_rente_wartezeit_5: bool,
+) -> bool:
+    """
+    Determine eligibility Erwerbsminderungsrente
+    (pension for reduced earning capacity)
+
+    Requirements are:
+    1 - not beeing able to work more than 3 hours per day (erwerbsgemindert)
+    2 - beeing insured for at least 5 years (allgemeine Wartezeit)
+    3 - having at least 3 years of mandatory contributions within the
+        last 5 years before retirement (the function below is just checking if
+        the amount of mandatory contribution months is high enough, not
+        if they took place within the last 5 years)
+
+    Legal reference: § 43 Abs. 1  SGB VI
+
+    Parameters
+    ----------
+    erwerbsgemindert
+        See basic input variable :ref:`erwerbsgemindert <erwerbsgemindert>.
+    m_pflichtbeitrag
+        See basic input variable :ref:`m_pflichtbeitrag <m_pflichtbeitrag>.
+    ges_rente_wartezeit_5
+        See :func:`ges_rente_wartezeit_5`.
+    Returns
+    -------
+    Erwerbsminderungsrente pension claim
+    """
+
+    anspruch_erwerbsm_rente = (
+        erwerbsgemindert and ges_rente_wartezeit_5 and m_pflichtbeitrag >= 36
+    )
+
+    return anspruch_erwerbsm_rente
+
+
+@dates_active(start="2004-01-01")
+def entgeltp_erwerbsm_rente(
+    entgeltp: float,
+    durchschn_entgelpt: float,
+    erwerbsm_rente_params: dict,
+    jahr_renteneintr: int,
+    age_of_retirement: float,
```

Comment:
Seems like it was not introduced here, but seeing 

```py
    jahr_renteneintr: int,
    age_of_retirement: float,
```

next to each other makes me wonder why we use different languages here?

And why would that not be just current age? Because it will differ over the course of the year? Touches on a discussion we just had with @TeBackh -- I don't think we can try to calculate `erwerbsminderungsrente` in 2010 for someone who took it up in 2003.

Reply:
This goes back to what we discussed [here](https://github.com/iza-institute-of-labor-economics/gettsim/issues/629#issuecomment-1665713136) I think. The Zurechnungszeit depends on age but also the date of the pension start since different amounts of Zurechungszeit are granted depending on whether the pension started before 2014, between 2014 and 2018, between 2018 and 2019, and after 2019. 

I agree on the naming issue, I guess since these variables are not directly institutional names I would suggest going with English?

Reply:
'jahr_renteneintr' is a basic input variable. Since all input variables have german names I think it would be better to switch to german instead.

Reply:
Yes, German sounds better.

But I still don't think we'll need that. What the function tries to do, say, is to calculate for the policy year 2023 the disability insurance benefit of somebody who took it up in 2015. We won't be able to do that consistently. In the policy year 2023, we can only reliably calculate the pension of people who retire in 2023.

The way to go about these dynamic settings will be to call GETTSIM repeatedly for different policy years. Say, if first year is 2015:

- 2015 for everyone
- 2016 once for people who took up DI pension in 2015 (pension level fixed); once for everyone else
- 2017 once for people who took up DI pension before 2017 (pension level fixed); once for everyone else
- ...

Reply:
I am not sure I understand why we will not be able to do this consistently. If we have information on the retirement date, age at retirement, and pension points, this should be enough to compute entitlements. The only thing that will change over time with the policy environment is the Rentenwert. 

Can you maybe elaborate a bit more?

Reply:
The thing is that we want to get rid of some functions when they become irrelevant. Think of `berufsunfähigkeitsrente` being implemented already. Will be gone from 2001 onwards as per `@dates_active`.

So this would work for the current use case, but it is not future-proof.

Reply:
Is it possible to find another workaround? If I understand correctly, this would mean we can only evaluate policies that affect retirees before or at the start of their pension, right? 

I think that would be quite unfortunate. For instance, there is an upcoming reform for 2024 that will increase disability benefits for individuals who didn't benefit from the reforms in the past decade. I am not sure how we would be able to implement that reform without having access to the age and date of retirement.

Reply:
Of course that will be possible! 

In general, there will be a little more work on the user side (=call GETTSIM repeatedly), that's all.

If there is a new reform that requires something like that as a state variable, we will need it only from 2024. Will be interesting to see how "individuals who didn't benefit from the reforms in the past decade" shall be identified...

Reply:
Maybe we should leave that change in the structure to a new PR, as it seems to affect many other components of the current model (all non em pensions ). 
Also wondering whether for now, where GETTSIM starts only in 1984 for most policies it would be preferrable to have at least a good approximation for pensioneers from earlier years rather than nothing.

Reply:
But those are unrelated changes, except they all are in the context of pensions? No change in structure required elsewhere in this PR. We should tackle those eventually, but not here.

Anything before 2005 needs a serious look, anyhow -- I don't think GETTSIM is of much use currently for that period. 1984 as a starting date should mostly be a placeholder, maybe someone put it into IZAΨMOD because the SOEP started at that time. 

Not sure what you mean with approximations in this context?

Reply:
> Not sure what you mean with approximations in this context?

I mean that the currently implemented pension thresholds are mostly true also for people who retired before 1984 (as we are testing them for cohorts  from 1900 with this sheet from the seibold paper). So we would be able to calculate pretty precise pensions for people who retired before 1984.

Reply:
Yes, sure, but what speaks against calling GETTSIM repeatedly then? Just use years before 1984, if the only output you request are pensions and all state variables are there, that should be no problem. Put differently, the starting date for 1984 in this case seems arbitrary and it seems like it can be set to an earlier value.

Reply:
Thats right. We should do this. 

Still somehow wondering whether we want the user to call gettsim 35 times when they have people with retirement entry years across a range of 35 years. but yeah thats how it is then

Reply:
Just a simple loop -- whether you call it twice or 35 times doesn't matter :smile: 

But yes, there is no denying that the origins of GETTSIM are in cross-sectional work. As we discussed in person the other day, given this structure there simply is no way of scaling dynamic stuff up as a one-stop experience.

Put differently, to keep things manageable, we always depict a policy environment as it is active at one particular date. Trying to achieve calculating the pension for someone who retired years ago would change this and it is simply not maintainable in the medium run.

Reply:
Maybe last note on this (perhaps obvious): We can only take over the zugangfsfaktor and Engeltpunkte from the year of retirement to subsequent years as state vars and we need to "freshly" calc the actual pension for every output year also for people who retired years ago, as the Rentenwert changes every year.

Reply:
We were discussing this among us Berliners and are wondering why we shouldn't implement it in a cross sectional way. Would it be possible to have a Zoom Call next week on this and discuss that? Also I agree with @TeBackh earlier comment, that we should handle this in a different PR.

Reply:
What do you mean by "implementing it in a cross-sectional way"? That's how I'd think of what I am suggesting. 

Again, it works for a simple case. But if functions cease to exist, you need to carry them around for as long as somebody might still be alive even if they retired 70 years ago. No loss in functionality, just a little more work for the user (which will be much less under updated interface).

Reply:
Early Monday morning (finish by 10am) would work for a call, I am blocked with a coauthor for most of the remainder of the week.

Reply:
> What do you mean by "implementing it in a cross-sectional way"? That's how I'd think of what I am suggesting.
> 
> Again, it works for a simple case. But if functions cease to exist, you need to carry them around for as long as somebody might still be alive even if they retired 70 years ago. No loss in functionality, just a little more work for the user (which will be much less under updated interface).

If you call it from 2015 up 2023 to calculate pension for someone who retired 2015, you will need to have the person in the dataset for all these years, i.e. a panel structure.Therefore I called it cross-sectional. But maybe I am just understanding it in the wrong way. 



> Early Monday morning (finish by 10am) would work for a call, I am blocked with a coauthor for most of the remainder of the week.

I am in the US for a research stay now, so that will be too early for me. Maybe we can arrange that in a quieter week and postpone these changes until then and do these in a separate PR.

Reply:
I really do not want to merge something that is due to be changed very soon again. I find the current structure simply too opaque to be able to review it in finite time, I am sorry. 

This is just about simplification, not about any changes that would take serious time.

Reply:
> If you call it from 2015 up 2023 to calculate pension for someone who retired 2015, you will need to have the person in the dataset for all these years, i.e. a panel structure.Therefore I called it cross-sectional. But maybe I am just understanding it in the wrong way.

Oh yeah, I was thinking from a GETTSIM perspective, you were thinking from a data perspective. All good.

Again, that case is supported, just a little bit of extra work. Just call GETTSIM for 2015 with the relevant variables measured *for the year 2015* (you just happen to label the data 2023, but they really are for the year of retirement).

Just to expand from @TeBackh and my conversations on this: In the structure you envision, we just do not see a way to consistently deal with pension credits earned *after* the first time a pension was drawn. You'd have to keep track of the entire history between 2015 and 2023.